### PR TITLE
Store request block hash in poc receipt

### PR DESCRIPTION
**Pull request by Vagabond**
Depends on helium/blockchain-core#622

### CODE REVIEW

File: test/miner_poc_SUITE.erl [hunk 1]
```diff
@@ -70,7 +70,7 @@ all() ->
      poc_dist_v8_test,
      poc_dist_v8_partitioned_test,
      poc_dist_v8_partitioned_lying_test,
-     poc_dist_v9_test,
+     poc_dist_v10_test,
```
Comment (vihu) [inline test/miner_poc_SUITE.erl:73]: I seemed to have missed adding the other two v9 tests here, can you please add them for v10?

Comment (evanmcc) [reply test/miner_poc_SUITE.erl:73]: while you're in here, can we comment out the v7 tests?

File: src/poc/miner_poc_statem.erl [hunk 2]
```diff
@@ -70,6 +70,7 @@
     responses = #{},
     receiving_timeout = ?RECEIVING_TIMEOUT :: non_neg_integer(),
     poc_hash  :: binary() | undefined,
+    request_block_hash  :: binary() | undefined,
```
Comment (evanmcc) [inline src/poc/miner_poc_statem.erl:73]: default this to `<<>>`?  otherwise we'll get a lot of crashes on the transition.  might never see it, though.

**Review by evanmcc — APPROVED** [review]
other than the nits it looks good

File: src/poc/miner_poc_statem.erl (hunk 2 above)
Comment (Vagabond) [reply src/poc/miner_poc_statem.erl:73]: I'm not sure how that will help?

**Pull request merged**